src/error.rs

```diff
@@ -61,59 +41,33 @@ impl Display for FictError {
     fn fmt(&self, f: &mut Formatter) -> Result<(), FmtError> {
         match *self {
             Message(ref s) => f.write_str(s),
-            IO(ref e) => Display::fmt(e, f),
-            Database(ref e) => Display::fmt(e, f),
-            Iron(ref e) => Display::fmt(e, f),
-            Hyper(ref e) => Display::fmt(e, f),
-            JSONDecode(ref e) => Display::fmt(e, f),
-            JSONEncode(ref e) => Display::fmt(e, f),
-            JSONParser(ref e) => Display::fmt(e, f),
+            Cause(ref e) => Display::fmt(e, f),
         }
     }
 }
 
-impl From<IOError> for FictError {
-    fn from(err: IOError) -> FictError {
-        FictError::IO(err)
+impl From<FmtError> for FictError {
+    fn from(err: FmtError) -> FictError {
+        FictError::Message(format!("{}", err))
     }
 }
 
-impl From<PgError> for FictError {
-    fn from(err: PgError) -> FictError {
-        FictError::Database(err)
-    }
-}
-
-impl From<IronError> for FictError {
-    fn from(err: IronError) -> FictError {
-        FictError::Iron(err)
-    }
-}
-
-impl From<HttpError> for FictError {
-    fn from(err: HttpError) -> FictError {
-        FictError::Hyper(err)
-    }
-}
-
-impl From<JSONDecoderError> for FictError {
-    fn from(err: JSONDecoderError) -> FictError {
-        FictError::JSONDecode(err)
-    }
-}
-
-impl From<JSONEncoderError> for FictError {
-    fn from(err: JSONEncoderError) -> FictError {
-        FictError::JSONEncode(err)
-    }
-}
-
-impl From<JSONParserError> for FictError {
-    fn from(err: JSONParserError) -> FictError {
-        FictError::JSONParser(err)
+trait NonFictError: Error {}
+impl NonFictError for io::Error {}
+impl NonFictError for iron::IronError {}
+impl NonFictError for postgres::Error {}
+impl NonFictError for hyper::HttpError {}
+impl NonFictError for rustc_serialize::json::DecoderError {}
+impl NonFictError for rustc_serialize::json::EncoderError {}
+impl NonFictError for rustc_serialize::json::ParserError {}
+
+impl<E: NonFictError + 'static> From<E> for FictError {
```

Comment:
I tried to implement this for all errors at once, like so:

```rust
impl<E: Error + 'static> From<E> for FictError {}
```

But because `FictError` *also* implements the `Error` trait, this conflicts with the `impl <T> From<T> for T` definition in core. Ah well.